src/poke_env/player/player.py

```diff
@@ -213,6 +213,15 @@ async def _create_battle(self, split_message: List[str]) -> AbstractBattle:
                         save_replays=self._save_replays,
                     )
 
+                # Add our team as teampreview_team, as part of battle initialisation
+                if isinstance(self._team, ConstantTeambuilder):
```

Comment:
is there data in here that we don't get during teampreview?

Reply:
Yes! It is primarily HP stats and nature (which interacts with berries). Though now that I look again at the teampreview request, you can get HP stats from "condition", and I think berry interactions are probably too niche and rare. I'm thinking I should simplify and remove?

Reply:
i think it makes sense to store this somewhere, but i'm not sure this is the right way to do it. the main change i'd like is instead of relying of the specifics of the constant teambuilder, store the generated team for each battle and inject it in the battle on teampreview requests
that being said, this implementation is a good first step - maybe we keep this as a first version and if we want to improve how we deal with this in the future we refactor it? i think this would also cover most cases